modules/navier_stokes/test/tests/finite_volume/materials/flow_diode/transient_operation.i

```diff
@@ -0,0 +1,307 @@
+    dx = '1 0.3 1'
+    dy = '0.5 0.2 0.5'
+    ix = '${nx} ${fparse nx/2} ${nx}'
+    iy = '${ny} ${ny} ${ny}'
+    subdomain_id = '1 1 1
+                    2 1 2
+                    3 4 1'
+  []
+
+  [add_walls]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = 'cmg'
+    primary_block = '1 3 4'
+    paired_block = '2'
+    new_boundary = 'walls'
+  []
+  [remove_wall_blocks]
+    type = BlockDeletionGenerator
+    input = add_walls
+    block = 2
+  []
+
+  # Add inlets and outlets
+  [top_left]
+    type = ParsedGenerateSideset
+    input = remove_wall_blocks
+    combinatorial_geometry = 'x<0.001 & y>0.6'
+    new_sideset_name = top_left
+  []
+  [bottom_left]
+    type = ParsedGenerateSideset
+    input = top_left
+    combinatorial_geometry = 'x<0.001 & y<0.6'
+    new_sideset_name = bottom_left
+  []
+  [top_right]
+    type = ParsedGenerateSideset
+    input = bottom_left
+    combinatorial_geometry = 'x>2.299 & y>0.6'
+    new_sideset_name = top_right
+  []
+  [bottom_right]
+    type = ParsedGenerateSideset
+    input = top_right
+    combinatorial_geometry = 'x>2.299 & y<0.6'
+    new_sideset_name = bottom_right
+  []
+
+  # Extra surfaces
+  [diode_inlet]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = bottom_right
+    primary_block = 4
+    paired_block = 3
+    new_boundary = 'diode_inlet'
+  []
+  [mid_section]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = diode_inlet
+    primary_block = 4
+    paired_block = 1
+    new_boundary = 'mid_connection'
+  []
+
+  [reduce_blocks]
+    type = RenameBlockGenerator
+    input = 'mid_section'
+    old_block = '4 3 1'
+    new_block = '1 diode fluid'
+  []
+[]
+
+[GlobalParams]
+  rhie_chow_user_object = 'pins_rhie_chow_interpolator'
+  advected_interp_method = 'upwind'
+  velocity_interp_method = 'rc'
+[]
+
+[Modules]
+  [NavierStokesFV]
+    compressibility = 'incompressible'
+    porous_medium_treatment = true
+
+    density = ${rho}
+    dynamic_viscosity = ${mu}
+
+    initial_velocity = '1e-6 1e-6 0'
+    initial_pressure = 0.0
+
+    inlet_boundaries = 'bottom_left top_right'
+    momentum_inlet_types = 'fixed-velocity fixed-velocity'
+    momentum_inlet_function = '1 0; -1 0'
+
+    wall_boundaries = 'top bottom walls'
+    momentum_wall_types = 'noslip noslip noslip'
+
+    outlet_boundaries = 'bottom_right top_left'
+    momentum_outlet_types = 'fixed-pressure fixed-pressure'
+    pressure_function = '1 1'
+
+    friction_blocks = 'fluid; diode'
+    friction_types = 'darcy forchheimer; darcy forchheimer'
+    # Base friction
+    # friction_coeffs = 'Darcy Forchheimer; Darcy Forchheimer'
+    # Combined with diode
+    friction_coeffs = 'combined_linear combined_quadratic; combined_linear combined_quadratic'
+
+    # Porosity jump treatment
+    # Option 1: diffusion correction
+    use_friction_correction = true
+    consistent_scaling = 10
+
+    # Option 2: bernouilli jump
+    # porosity_interface_pressure_treatment = bernoulli
```

Comment:
Do you want to mention porosity smoothing if you're going to list options?

Reply:
I kind of mean it as advertisement, and if we want to try something else here